### PR TITLE
Implement endorse action (#371)

### PR DESCRIPTION
**Pull request by micahstairs**
Screen Shot 2022-12-28 at 8 18 39 AM
Screen Shot 2022-12-29 at 4 43 10 AM

NOTE: These screenshots were taken from separate games, so the game state was a bit different.

### CODE REVIEW

File: innovation.game.php [hunk 1]
```diff
@@ -7315,6 +7340,52 @@ function dogma($card_id) {
...
+        // Check that the player really has this card on his board
+        if ($card_to_endorse['owner'] != $player_id || $card_to_endorse['location'] != "board") {
+            self::throwInvalidChoiceException();
+        }
+        // Card is not at the top of a stack
+        if (!self::isTopBoardCard($card_to_endorse)) {
+            self::throwInvalidChoiceException();
+        }
+        // City cards and Battleship Yamato do not have a featured icon so they cannot be executed as a dogma effect
+        if ($card_to_endorse['dogma_icon'] == null) {
```
Comment (ultimatefiend) [inline innovation.game.php:7364]: https://boardgamegeek.com/thread/1815267/dogma-actions-and-cities

Cities can be endorsed

Comment (micahstairs) [reply innovation.game.php:7364]: Thanks, fixed!

**Comment by micahstairs** [comment]
Screen Shot 2022-12-29 at 9 38 42 PM

**Review by ultimatefiend — COMMENTED** [review]
See comment about cities should be endorsable (and dogmable)

**Comment by micahstairs** [comment]
> See comment about cities should be endorsable (and dogmable)

Yup! I already made that change about an hour ago. I attached a screenshot of the dogma, but the endorsement should work fine too.

**Review by ultimatefiend — APPROVED** [review]
do city cards always warn if someone tries to dogma them?  (like a demand when you don't meet the criteria)

**Comment by micahstairs** [comment]
> do city cards always warn if someone tries to dogma them? (like a demand when you don't meet the criteria)

Ah I forgot to do that. Just pushed a commit to add this warning, thanks!

**Pull request merged**